.github/workflows/ci.yml

```diff
@@ -7,41 +7,25 @@ on:
   pull_request:
 
 jobs:
-  assemble:
-
+  build:
     runs-on: ubuntu-latest
     timeout-minutes: 15
-    
-    steps:
-      # These setup steps should be common across all jobs in this workflow.
-      - uses: actions/checkout@v2
-      - uses: gradle/wrapper-validation-action@v1
-      - name: Set up JDK 1.8
-        uses: actions/setup-java@v1
-        with:
-          java-version: 1.8
-
-      - name: Assemble with Gradle
-        run: ./gradlew assemble --no-build-cache --no-daemon --stacktrace
 
-  # Runs all check tasks in parallel.
-  check:
-    name: Check
-    needs: assemble
-    runs-on: ubuntu-latest
-    timeout-minutes: 10
     strategy:
-      # Run all checks, even if some fail.
+      # Run all tasks, even if some fail. Note that they don't share an output, some tasks overlap
+      # which is expected. If they need to share their outputs, then we need a proper caching
+      # solution.
       fail-fast: false
       matrix:
         gradle-task:
-          # Unit tests
+          - assemble
```

Comment:
I think these can all be combined by just running `build`.

Reply:
It also runs `lint`, which I want to have separate.